DICOMPlugins/DICOMTID1500Plugin.py

```diff
@@ -157,6 +292,899 @@ def getDateTime(self, uid):
+    else:
+      checkUsePlugin = 1 
+
+    return checkUsePlugin
+  
+  def checkIfSRContainsBbox(self, srFileName):
+    """
+    Checks if the SR contains a bounding box or not. 
+    """
+
+    # default is that SR does not contain a bounding box 
+    checkIfSRContainsBbox = 0 
+
+    # read the SR using highdicom 
+    sr = hd.sr.srread(srFileName)
+
+    # First get the planar roi measurement groups 
+    groups = sr.content.get_planar_roi_measurement_groups()
+    num_groups = len(groups)
+    # Then we check if the SR contains a bounding box or not 
+    for group in groups: 
+      qual_evals = group.get_qualitative_evaluations()
+      for qual_eval in qual_evals: 
+        if qual_eval.name.CodeValue == "130400" and \
+          qual_eval.name.CodingSchemeDesignator == "DCM" and \
+          qual_eval.name.CodeMeaning == "Geometric purpose of region" and \
+          qual_eval.value.CodeValue == "75958009" and \
+          qual_eval.value.CodingSchemeDesignator == "SCT" and \
+          qual_eval.value.CodeMeaning == "Bounded by": 
+          checkIfSRContainsBbox = 1 
+        else: 
+          checkIfSRContainsBbox = 0 
+
+    return checkIfSRContainsBbox 
+  
+  def checkIfSRContainsPoint(self, srFileName): 
+    """
+    Checks if the SR contains a point. 
+    """
+
+    # default is that the SR does not contain a point 
+    checkIfSRContainsPoint = 0 
+
+    # read the SR using highdicom 
+    sr = hd.sr.srread(srFileName)
+
+    # create the ImageRegion3D code 
+    image_region_code = hd.sr.value_types.Code(
+        value='111030',
+        scheme_designator='DCM',
+        meaning='Image Region'
+    )
+
+    # First get the planar roi measurement groups 
+    groups = sr.content.get_planar_roi_measurement_groups()
+    num_groups = len(groups)
+
+    # Then we check if the SR contains a point or not 
+    # Iterate through the groups 
+    for group in groups: 
+      # group.reference_type = Code(value='111030', scheme_designator='DCM', meaning='Image Region', scheme_version=None)
+      # Check if there is a reference_type, should be ImageRegion
+      try: 
+        reference_type = group.reference_type
+      except: 
+        print('reference_type does not exist for group')
+      # If it does equal image_region_code, then check if it's a POINT
+      if (reference_type == image_region_code):
+        try: 
+          # type(group.roi) = highdicom.sr.content.ImageRegion3D
+          graphic_type = group.roi.GraphicType
+        except: 
+          print('GraphicType does not exist for group.roi')
+        if (graphic_type == "POINT"): 
+          checkIfSRContainsPoint = 1 
+ 
+    return checkIfSRContainsPoint 
+  
+  def checkIfSRContainsPolyline(self, srFileName): 
+    """
+    Checks if the SR contains a polyline. 
+    """
+
+    # default is that the SR contains a polyline 
+    checkIfSRContainsPolyline  = 0 
+
+    # read the SR using highdicom 
+    sr = hd.sr.srread(srFileName)
```

Comment:
You are reading the source SR multiple times - can you read once and access via a variable?